src/planner/binder/statement/bind_create.cpp

```diff
@@ -635,17 +635,21 @@ BoundStatement Binder::Bind(CreateStatement &stmt) {
 		string source_path = base.path;
 
 		auto &config = DBConfig::GetConfig(context);
-		for (auto &extension : config.create_database_extensions) {
-			auto create_database_function_ref =
-			    extension.function(context, extension_name, database_name, source_path, extension.data.get());
-			if (create_database_function_ref) {
-				auto bound_create_database_func = Bind(*create_database_function_ref);
-				result.plan = CreatePlan(*bound_create_database_func);
-				break;
-			}
-		}
-		if (!result.plan) {
+
+		// for now only handling the case where there is one storage extension registered
+		if (config.storage_extensions.empty()) {
 			throw NotImplementedException("CREATE DATABASE not supported in DuckDB yet");
+		} else if (config.storage_extensions.size() > 1) {
+			throw BinderException("CREATE DATABASE only supported when there is one storage extension registered.");
+		}
+		auto entry = config.storage_extensions.begin();
+		auto &storage_extension = entry->second;
+		auto create_database_function_ref = storage_extension->create_database_extension_function(
```

Comment:
The `create_database_extension_function` might be a nullptr (meaning the extension does not support create db). We should handle this case

Reply:
Done